### PR TITLE
Add Binary and Hashable instances.

### PR DESCRIPTION
**Pull request by locallycompact**

### CODE REVIEW

File: composite-binary-instance/package.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name:                composite-binary-instance
```
Comment (Dridus) [inline composite-binary-instance/package.yaml]: can you rename this to `composite-binary` to match the overall naming scheme?

File: composite-binary-instance/package.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  pname = "composite-binary-instance";
+  version = "0.7.2.0";
+  src = ./.;
+  libraryHaskellDepends = [
+    aeson-better-errors base composite-aeson mtl refined
+  ];
+  libraryToolDepends = [ hpack ];
+  prePatch = "hpack";
+  homepage = "https://github.com/ConferOpenSource/composite#readme";
+  description = "composite-aeson support for Refined from the refined package";
```
Comment (Dridus) [inline composite-binary-instance/package.nix]: same comment here about using `make` to have this file auto generated

File: composite-binary-instance/src/Composite/Record/Binary.hs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{-# LANGUAGE UndecidableInstances #-}
+module Composite.Record.Binary where
+
+import Composite.Record((:->), Record, pattern (:*:), val, getVal)
+import Control.Applicative(liftA2)
+import Data.Binary(Binary(put, get))
+import Data.Functor.Identity(runIdentity)
+
+instance Binary a => Binary (s :-> a) where
```
Comment (Dridus) [inline composite-binary-instance/src/Composite/Record/Binary.hs]: `deriving instance Binary a => Binary (s :-> a)` with `GeneralizedNewtypeDeriving` turned on should do the same thing?

File: composite-binary-instance/src/Composite/Record/Binary.hs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  put = put . getVal
+  get = fmap (runIdentity . val) get
+
+instance Binary (Record '[])
+
+instance (Binary a, Binary (Record xs), x ~ (s :-> a)) => Binary (Record (x : xs)) where
```
Comment (Dridus) [inline composite-binary-instance/src/Composite/Record/Binary.hs]: same comment here as for `NFData` - make it more general, e.g. `instance (Binary (f r), Binary (Rec f rs)) => Binary (Rec f (r ': rs))` and also make sure there are no compiler warnings being eaten

File: stack.yaml [hunk 5]
```diff
@@ -3,6 +3,7 @@ packages:
   - composite-aeson
   - composite-aeson-refined
   - composite-base
+  - composite-binary-instance
```
Comment (Dridus) [inline stack.yaml]: and to `stack-8.8.3.yaml`?

File: composite-hashable-instance/package.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
+name:                composite-hashable-instance
```
Comment (Dridus) [inline composite-hashable-instance/package.yaml]: same request with naming - drop `-instance` please

File: composite-hashable-instance/src/Composite/Record/Hashable.hs [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import Composite.Record((:->), Record, Rec(RNil), pattern (:*:), val, getVal)
+import Data.Hashable(Hashable(hashWithSalt))
+
+instance Hashable a => Hashable (s :-> a) where
+  hashWithSalt n x = hashWithSalt n $ getVal x
+
+instance Hashable (Record '[]) where
+  hashWithSalt n RNil = n `hashWithSalt` ()
+
+instance (Hashable a, Hashable (Record xs), x ~ (s :-> a)) => Hashable (Record (x : xs)) where
```
Comment (Dridus) [inline composite-hashable-instance/src/Composite/Record/Hashable.hs]: same comment about generalizing as for `NFData` and `Binary`

**Review by Dridus — CHANGES_REQUESTED** [review]
since you have all the commits stacked, feel free to submit a unified PR for all four issues

**Comment by locallycompact** [comment]
The whole stack is here, lmk if I missed anything.

**Review by Dridus — CHANGES_REQUESTED** [review]
`composite-aeson-path/src/Composite/Aeson/Path.hs` is triggering the orphan warning/error, so needs `{-# OPTIONS_GHC -fno-warn-orphans #-}`

**Comment by locallycompact** [comment]
How's this?

**Review by Dridus — APPROVED** [review]

**Pull request merged**